app/electron/main.ts

```diff
@@ -383,6 +402,63 @@ function startElecron() {
     });
 
     if (!useExternalServer) {
+      const runningHeadlamp = await getRunningHeadlampPIDs();
+      if (!!runningHeadlamp) {
+        const resp = dialog.showMessageBoxSync(mainWindow, {
+          // Avoiding mentioning Headlamp here because it may run under a different name depending on branding (plugins).
+          title: i18n.t('Other process running'),
+          message: i18n.t(
+            'Looks like another process is already running. Continue by stopping that process automatically, or quit?'
+          ),
+          type: 'question',
+          buttons: [i18n.t('Continue'), i18n.t('Quit')],
+        });
+
+        if (resp === 0) {
+          runningHeadlamp.forEach(killProcess);
+        } else {
+          mainWindow.close();
+          return;
+        }
+      }
+
+      // If we reach here, then we attempted to kill headlamp-server. Let's make sure it's killed
+      // before starting our own, or else we may end up in a race condition (failing to start the
+      // new one before the existing one is fully killed).
+      if (!!runningHeadlamp) {
+        let stillRunning = true;
+        const waitTime = 250; //ms
+        const attempts = (10 * 1000) / 250; // 4 times per sec, 10 secs max
+        for (let tries = 0; tries < attempts && stillRunning; tries++) {
+          console.debug(
+            `Checking if Headlamp is still running after we asked it to be killed; try=${
+              tries + 1
+            }/${attempts}`
+          );
+          // Wait 250ms
+          await setTimeout(() => {}, waitTime);
+
+          stillRunning = !!(await getRunningHeadlampPIDs());
+          console.debug(stillRunning ? 'Still running...' : 'No longer running!');
+        }
+
+        // If we couldn't kill the process, warn the user and quit.
+        const processes = await getRunningHeadlampPIDs();
+        if (!!processes) {
+          dialog.showMessageBoxSync({
+            type: 'warning',
+            title: i18n.t('Failed to quit the other running process'),
+            message: i18n.t(
+              `Could not quit the other running process, PIDs: {{ process_list }}. Please stop that process and relaunch the app.`,
```

Comment:
Probably need to add these to the locale files.

How about mentioning the process name ("headlamp-server") in the message too?

> Could not quit the "headlamp-server" background process, ...

Reply:
I have avoided mentioning "headlamp" so users who completely override the branding don't get users confused. Not sure if this is a good exception or not...

Reply:
Yeah, ok. Probably better to avoid mentioning Headlamp.